packages/insured-bridge-relayer/src/Relayer.ts

```diff
@@ -721,4 +725,56 @@ export class Relayer {
         break;
     }
   }
+
+  // Return unique deposit event matching relay
+  private async matchRelayWithDeposit(relay: Relay): Promise<Deposit | undefined> {
+    // First try to fetch deposit from the L2 client's default block search config. This should work in most cases.
+    let deposit: Deposit | undefined = this.l2Client.getDepositByHash(relay.depositHash);
+
+    // We could not find a deposit using the L2 client's default block search config. Next, we'll modify the block
+    // searcg config using the relay's quote time. This allows us to capture any deposits that happened outside of
+    // the L2 client's default block search config.
+    if (deposit === undefined) {
+      const l2BlockForDepositQuoteTime = await findBlockNumberAtTimestamp(this.l2Client.l2Web3, relay.quoteTimestamp);
```

Comment:
for now this is likely fine though as in almost all cases this branch won't get hit

Reply:
I think its safer to use the contract's deployment time but one problem with this is that the max amount of blocks that we can query might be < `now - contract.deploymentTime` which means we'd need to add additional logic to break up this time period into chunks. Looking up the relay.quoteTime seems like it'd capture the deposit in all cases since the deposit box forces `quoteTime` to be +/- 10 minutes from the current time. So as long as the lookback is > 20 minutes long, it will capture it.